service/internal/cloudconfig/tccpn.go

```diff
@@ -40,6 +41,26 @@ func NewTCCPN(config TCCPNConfig) (*TCCPN, error) {
 	return t, nil
 }
 
+func (t *TCCPN) NewHashes(ctx context.Context, obj interface{}) ([]string, error) {
+	var hashes []string
+
+	templates, err := t.NewTemplates(ctx, obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	for _, t := range templates {
+		h := sha512.New()
+		_, err := h.Write([]byte("sha512-" + t))
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		hashes = append(hashes, string(h.Sum(nil)))
```

Comment:
We need to change it a bit and use `hex.EncodeToString`

```go
package main

import (
	"crypto/sha512"
	"encoding/hex"
	"fmt"
)

func main() {
	s := sha512.New()
	s.Write([]byte("hello sha world\n"))
	fmt.Println(string(s.Sum(nil)))
	fmt.Println(hex.EncodeToString(s.Sum(nil)))
}
```
This will print
```
����)u�����V-ߏ�����'�����r�|�QK��j�����&�
9c5b761a9bf8f74da84c57101669b8042b860ca5871a15c4e729758ad1dee58d562ddf8fa303ada1f3e0278fdff4af9a72f57ce0514be98f6a959daabda926f5
```

Without we will won't be able to create/update CF stacks
```
 "annotation": "ValidationError: Template contains invalid characters\n\tstatus code: 400, request id: 6a8da8fc-3edc-4479-a082-fc6dfb334998",
```

```suggestion
		hashes = append(hashes, hex.EncodeToString(s.Sum(nil)))
```